tools/sloths/src/utils/canvas-utils.ts

```diff
@@ -0,0 +1,541 @@
+        moveElement(mousePos, el, canvasScaleSteps);
+        isHovered = el.isHovered;
+      }
+    }
+  }
+};
+
+export const selectElement = (mousePos: CanvasPos, el: CanvasElement) => {
+  const canvasElement = el;
+
+  const elXY = getElRectXY(el);
+
+  const isLeftBorderSelected =
+    mousePos.x >= elXY.x1 && mousePos.x <= elXY.x1 + 2 && mousePos.y >= elXY.y1 && mousePos.y <= elXY.y2;
+
+  const isRightBorderSelected =
+    mousePos.x >= elXY.x2 - 2 && mousePos.x <= elXY.x2 && mousePos.y >= elXY.y1 && mousePos.y <= elXY.y2;
+
+  const isSelected = mousePos.x >= elXY.x1 && mousePos.x <= elXY.x2 && mousePos.y >= elXY.y1 && mousePos.y <= elXY.y2;
+
+  canvasElement.isLeftBorderSelected = isLeftBorderSelected;
+  canvasElement.isRightBorderSelected = isRightBorderSelected && !isLeftBorderSelected;
+  canvasElement.isSelected = isSelected && !isLeftBorderSelected && !isRightBorderSelected;
+
+  if (isSelected || isLeftBorderSelected || isRightBorderSelected) {
+    canvasElement.selectedPos = { ...mousePos };
+  }
+};
+
+export const selectElements = (e: MouseEvent, canvas: HTMLCanvasElement, layers: CanvasElement[]) => {
+  const mousePos = getMousePos(e, canvas);
+
+  let isSelected = false;
+
+  for (let i = layers.length - 1; i >= 0; i -= 1) {
+    const el = layers[i];
+
+    if (isSelected) {
+      el.isLeftBorderSelected = false;
+      el.isRightBorderSelected = false;
+      el.isSelected = false;
+    } else {
+      selectElement(mousePos, el);
+      isSelected = el.isSelected || el.isLeftBorderSelected || el.isRightBorderSelected;
+    }
+  }
+};
+
+export const deselectElements = (layers: CanvasElement[]) => {
+  layers.forEach((el) => {
+    const canvasElement = el;
+
+    canvasElement.isSelected = false;
+    canvasElement.isLeftBorderSelected = false;
+    canvasElement.isRightBorderSelected = false;
+    canvasElement.selectedPos = {} as CanvasPos;
+  });
+};
+
+export const unhoverElements = (layers: CanvasElement[]) => {
+  layers.forEach((el) => {
+    const canvasElement = el;
+
+    canvasElement.isHovered = false;
+    canvasElement.isBorderHovered = false;
+  });
+};
+
+// scaling
+export const scaleUpCanvas = (props: CanvasProperties) => {
+  const canvasProps = props;
+  canvasProps.scaleSteps = Math.min(canvasProps.scaleMax, canvasProps.scaleSteps + 0.02);
+};
+
+export const scaleTrueCanvas = (props: CanvasProperties, layers: CanvasElement[]) => {
+  const canvasProps = props;
+  canvasProps.scaleSteps = 1;
+
+  layers.forEach((el) => {
+    const canvasElement = el;
+    canvasElement.scaleSteps = canvasElement.scaleTrue;
+  });
+};
+
+export const scaleDownCanvas = (props: CanvasProperties) => {
+  const canvasProps = props;
+  canvasProps.scaleSteps = Math.max(canvasProps.scaleMin, canvasProps.scaleSteps - 0.02);
+};
+
+export const scalingElements = (e: WheelEvent, layers: CanvasElement[], props: CanvasProperties) => {
+  if (e.deltaY === 0) return;
+
+  const dStep = e.deltaY > 0 ? 0.05 : -0.05;
```

Comment:
Please consider to extract `0.05` in variable and give it meaningful name

Reply:
done